### ISSUE TITLE
[Permalink] Using the document's actual title instead of the its file name

### ISSUE THREAD

**Issue by wsmd** [jekyll/jekyll#4012]
Say I have this document named `2015-10-06-dont-show-this.md`

and in in its front matter I have this title

```YAML
---
title: this is the actual title
---
```

I've set the permalink in my `_config.yml` to `:categories/:title` and I understand that the documentation says:
> `title` => Title from the document’s filename. May be overridden via the document’s slug YAML front matter.

When the site renders of course the link to the page will be:
```
http://link.to.site/category/downt-show-this
```

Is there any way I can use the article's actual title in the link? So it would be :+1:
```
http://link.to.site/category/this-is-the-actual-title
```

Maybe something like auto slugging the actual title?

**Comment by mattr-** [comment]
I believe your answer is "hidden" in the docs you posted. 

> `title` => Title from the document’s filename. May be overridden via the document’s slug YAML front matter.

You'll need a piece of front matter named `slug` with the slug you actually want the post to have.

**Comment by wsmd** [comment]
@mattr- That's right. But I'm looking for an "automatic" way to do it without having to enter the slug manually. Simply put, Jekyll should use the title (front matter) of the article and automatically creates an a slug from it.
